crates/nargo_cli/src/cli/init_cmd.rs

```diff
@@ -24,33 +33,53 @@ fn test_main() {
 }
 "#;
 
+const LIB_EXAMPLE: &str = r#"fn my_util(x : Field, y : Field) -> bool {
+    x != y
+}
+
+#[test]
+fn test_main() {
+    assert(my_util(1, 2));
+
+    // Uncomment to make test fail
+    // assert(my_util(1, 1));
+}
+"#;
+
 pub(crate) fn run<B: Backend>(
     // Backend is currently unused, but we might want to use it to inform the "new" template in the future
     _backend: &B,
-    _args: InitCommand,
+    args: InitCommand,
     config: NargoConfig,
 ) -> Result<(), CliError<B>> {
-    initialize_project(config.program_dir);
+    let package_type = if args.lib { PackageType::Library } else { PackageType::Binary };
+    initialize_project(config.program_dir, package_type);
     Ok(())
 }
 
 /// Initializes a new Noir project in `package_dir`.
-pub(crate) fn initialize_project(package_dir: PathBuf) {
+pub(crate) fn initialize_project(package_dir: PathBuf, package_type: PackageType) {
     // TODO: Should this reject if we have non-Unicode filepaths?
     let package_name = package_dir.file_name().expect("Expected a filename").to_string_lossy();
     let src_dir = package_dir.join(SRC_DIR);
     create_named_dir(&src_dir, "src");
 
+    // TODO: Need to make type configurable
     let toml_contents = format!(
         r#"[package]
 name = "{package_name}"
+type = "{package_type}"
 authors = [""]
 compiler_version = "{CARGO_PKG_VERSION}"
 
 [dependencies]"#
     );
 
     write_to_file(toml_contents.as_bytes(), &package_dir.join(PKG_FILE));
-    write_to_file(EXAMPLE.as_bytes(), &src_dir.join("main.nr"));
-    println!("Project successfully created! Binary located at {}", package_dir.display());
+    // This is a match so we get a compile error when we add to the package types
```

Comment:
```suggestion
    // This uses the `match` syntax instead of `if` so we get a compile error when we add new package types (which likely need new template files)
```

Reply:
Ah, much clearer. Thank you.